### PR TITLE
ISPN-15166 SIFS rebuild index cannot compact file

### PR DESCRIPTION
**Pull request by wburns**
https://issues.redhat.com/browse/ISPN-15166

### CODE REVIEW

File: core/src/test/java/org/infinispan/persistence/sifs/SoftIndexFileStoreRestartTest.java [hunk 1]
```diff
@@ -238,4 +238,20 @@ public void testRestartWithEntryUpdatedMultipleTimes(boolean leafOrNode) throws
...
+         cache(0, cacheName).put("same-key", "value-" + i);
+      }
+
+      killMember(0, cacheName);
+
+      if (deleteIndex) {
+         // Delete the index which should force it to rebuild
+         Util.recursiveFileRemove(Paths.get(tmpDirectory, "index"));
+      }
+      createCacheManagers();
```
Comment (pruivo) [inline core/src/test/java/org/infinispan/persistence/sifs/SoftIndexFileStoreRestartTest.java:255]: could you check the key's value after restart?

Comment (wburns) [reply core/src/test/java/org/infinispan/persistence/sifs/SoftIndexFileStoreRestartTest.java:255]: That is already done in other tests. If you undo the other fix the teardown phase of the test will hang :(

Comment (wburns) [reply core/src/test/java/org/infinispan/persistence/sifs/SoftIndexFileStoreRestartTest.java:255]: I can rename the test though to better show what it is actually testing ;)

**Review by pruivo — APPROVED** [review]

**Comment by pruivo** [comment]
Merged! thanks @wburns !

**Pull request merged**